### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by tommyp**
Within certain Finders, we don't want the documents per page to be all of them (or 1000). This PR allows us to specify a `documents_per_page` as part of the content item and have Finder Frontend build the pagination query within the `SearchQueryBuilder` and render the Previous and Next component. [Ticket](https://trello.com/c/NfxTcDci/273-paginate-finder-frontend). Most of the work was completed by @dsingleton. I renamed some things and added tests.

Screenshot:
screen shot 2015-06-30 at 12 42 15

Tests will go green when https://github.com/alphagov/govuk-content-schemas/pull/91 is merged.

### CODE REVIEW

**Comment by dsingleton** [comment]
I've not had a proper look yet, but I don't think pagination will stay in sync with (leaving inaccurate pagination UI), which'll be confusing for users. IMO thats a blocker for merging this.

File: app/parsers/result_set_parser.rb [hunk 1]
```diff
@@ -1,6 +1,5 @@
 module ResultSetParser
   def self.parse(results, total, finder)
-
```
Comment (dsingleton) [inline app/parsers/result_set_parser.rb]: Should probably remove this, to keep the commit cleaner.

**Comment by dsingleton** [comment]
AJAX/pagination aside I think it looks good. Although someone else should review as I did a bit of work on it.

**Comment by tommyp** [comment]
Closing until we are able to do the pagination through AJAX.

**Pull request closed without merging**